lib/web/app/handler.go

```diff
@@ -136,6 +137,51 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	h.router.ServeHTTP(w, r)
 }
 
+// HandleConnection handles connections from plain TCP applications.
+func (h *Handler) HandleConnection(ctx context.Context, clientConn net.Conn) error {
+	tlsConn, ok := clientConn.(*tls.Conn)
+	if !ok {
+		return trace.BadParameter("expected *tls.Conn, got: %T", clientConn)
+	}
+
+	certs := tlsConn.ConnectionState().PeerCertificates
+	if len(certs) != 1 {
```

Comment:
Question: I'm guessing this is required to be 1 because we need to assert identity, and even if multiple valid certificates were to be presented, that should be considered an error condition because we wouldn't know which one to use? Will this cause any problems with applications that send client certificate chains?

Reply:
The request is coming from the Teleport local proxy which establishes an mTLS tunnel to the Proxy which is an application we control and we don't send certificate chains.